include/EVT/utils/types/FixedQueue.hpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * The FixedQueue is a circular FIFO queue that can hold up to a fixed
+ * sized number of elements. Each instance of the queue will have the same
+ * max capacity, but the user can provide an artifical limit.
+ *
+ * The queue will also be able to support overridding old data when the
+ * capactiy is reached.
+ *
+ * NOTE: The elements of the queue must implement the copy assignment
+ * operator.
+ *
+ * TODO: Consider locking mechanism around critical regions.
+ *
+ * @author Collin Bolles
+ */
+#ifndef _EVT_FIXED_QUEUE_
+#define _EVT_FIXED_QUEUE_
+
+#include <stdint.h>
+
+// The most number of elements in the queue, the user can provide a
+// value that is less then this number to reduce the apparent capacity.
+// Note that doing so will not reduce the size of the instance.
+#define QUEUE_MAX_CAPACITY 512
+
+namespace EVT::core::types
+{
+
+template<class Element>
+class FixedQueue {
```

Comment:
Big fan of this. Reminds me of something I did back at Magna (non-CAN related)! Definitely a better way of handling messages.

Reply:
Thanks! I was happy with how the queue based version worked out.